p2p/client.go

```diff
@@ -431,6 +532,87 @@ func (c *Client) bootstrapLoop(ctx context.Context) {
 	}
 }
 
+// latest seen height to be retrieved using block-sync
+func (c *Client) setLatestSeenHeight(height uint64) {
+	c.latestSeenHeight = max(height, c.latestSeenHeight)
+}
+
+// retrieveBlockSyncLoop checks if there is any missing block to be retrieved and requests it on demand
+func (c *Client) retrieveBlockSyncLoop(ctx context.Context) {
+	ticker := time.NewTicker(c.conf.BlockSyncRetrieveRetryTime)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			state, err := c.store.LoadState()
+			if err != nil {
+				c.logger.Error("loading state", "err", err)
+				continue
+			}
+
+			// this loop iterates and retrieves all the blocks between the last block applied and the greatest height received,
+			// skipping any block cached, since are already received.
+			for h := state.NextHeight(); h <= c.latestSeenHeight; h++ {
+				_, ok := c.blockAlreadyReceived[h]
+				if ok {
+					continue
+				}
+				c.logger.Debug("Getting block", "height", h)
+				bid, err := c.GetBlockId(ctx, h)
+				if err != nil || bid == cid.Undef {
+					c.logger.Error("unable to find cid", "height", h)
+					continue
+				}
+				_, err = c.store.SaveBlockCid(h, bid, nil)
+				if err != nil {
+					c.logger.Error("storing block cid", "height", h, "cid", bid)
+					continue
+				}
+				block, err := c.blocksync.GetBlock(ctx, bid)
+				if err != nil {
+					continue
+				}
+				if c.blocksync.msgHandler != nil {
+					c.blocksync.msgHandler(&block)
+				}
+			}
+			for h := range c.blockAlreadyReceived {
+				if h < state.NextHeight() {
+					delete(c.blockAlreadyReceived, h)
+				}
+			}
+		}
+	}
+}
+
+func (c *Client) advertiseBlockSyncCids(ctx context.Context) {
+	state, err := c.store.LoadState()
+	if err != nil {
+		c.logger.Error("loading state", "err", err)
+		return
+	}
+	for h := state.BaseHeight; h <= state.Height(); h++ {
+		id, err := c.GetBlockId(ctx, h)
```

Comment:
IMO quite confusing
not sure what the difference between
`id, err := c.GetBlockId`
and
`id, err = c.store.LoadBlockCid(h)`

Reply:
GetBlockId gets the block id from the network (DHT) - in case not known - and  c.store.LoadBlockCid it gets from local store. i renamed to GetBlockIdFromDHT to make it clear.